src/sys/wait.rs

```diff
@@ -235,6 +235,57 @@ pub fn waitpid<P: Into<Option<Pid>>>(pid: P, options: Option<WaitPidFlag>) -> Re
     }
 }
 
+
+/// The same as calling waitpid(None, None)
 pub fn wait() -> Result<WaitStatus> {
     waitpid(None, None)
 }
+
+
+/// Returns an iterator over all currently available events of child processes.
+/// 
+/// This iterator is best used with a signal handler on the signal SIGCHLD
+/// (Note that this example uses the signal_hook crate. This is just for clarity, it would of course work with any other mechanism to
+/// receive signals)
+/// ```ignore
+/// for signal in signals.forever() {
+///     match signal as libc::c_int {
+///         signal_hook::SIGCHLD => {
+///             // one or more child events exist. We need to handle all now, because the signal is not sent for each.
+///             child_event_iter
+///                 .take_while(Result::is_ok) // you probably want to do some error handling, instead of just stopping the iterator
+///                 .for_each(|val| {
+///                     println!("A child did something! {:?}", val);
+///                 });
+///         }
+///         _ => unreachable!("We only handle SIGCHLD"),
+///     }
+/// }
+/// ```
+pub fn child_event_iter() -> ChildEventIter {
+    std::iter::from_fn(get_next_child_event)
+}
+type ChildEventIter = std::iter::FromFn<fn() -> std::option::Option<Result<WaitStatus>>>;
+
+/// use waitpid with WNOHANG to poll the next event or return None if no more events exist
+fn get_next_child_event() -> Option<Result<WaitStatus>> {
+    match waitpid(None, Some(WaitPidFlag::WNOHANG)) {
+        Ok(status) => Some(Ok(status)),
+        Err(e) => {
+            if let crate::Error::Sys(crate::errno::Errno::ECHILD) = e {
+                // No more events to be waited on
+                None
+            } else {
+                // Some other error, which we cannot handle here
+                //
+                // This might be unecessary, since the other errors are either
+                // 1. EINTR which only occurs if WNOHANG is not used, but it is used here
+                // 2. EINVAL which should be impossible since only valid WaitPidFlags can be created
+                //
+                // Not having this error handling would make the API nicer, but it could introduce breakage on some
```

Comment:
It's a good idea to have the error handling.  Plenty of syscalls occasionally return undocumented errors.